### DISCUSSION TITLE
Disable git hook installation?

### DISCUSSION THREAD

**Discussion by drupol** [phpro/grumphp#1017] [Ideas]
Hi there,

I'm wondering if disable git hook installation through a config parameter is a good idea.

Sometimes we work on a project where git is not enabled and installing Grumphp fails.
Sometimes we just don't want to have the git hook installed by default even if Git is enabled...

Maybe we should have a configuration option that would let the user choose the behavior ?

I know that [it is already possible to do](https://github.com/phpro/grumphp/blob/master/doc/installation/exotic.md#register-paths-in-your-composer-file), but wouldn't it be better to have this setting in the Grumphp config file ?

Something like:
```yaml
parameters:
  install_git_hook: true
```

Available values for this property could be:
* `true` : Enable the Git hook if git is enabled on the project, fails otherwise (default current behavior)
* `false`: Do not enable the Git hook
* `null`: Enable the Git hook is git is enabled on the project, skip otherwise and stay silent

Let me know what you think!

**Comment by veewee** [comment]
Why would you  install grumphp on projects that don't use git?
It is a hard dependency.

The reason why it currently is not in grumphp.yaml, is because the decision is made in the composer plugin.
We don't read the grumphp file at that moment.

Not sure if it's a good idea to make it configurable at 2 different locations.

**Reply by drupol** [reply]
I personally never had the issue because I'm always using git in any PHP project.

Despite that Git it is a hard dependency, Grumphp is often used as a task launcher and in the CI while the git hook part is usually skipped.

I always suggest people to disable the Git hook installation using the composer.json file, but as we are using a centralized config for Grumphp, I was wondering if this was possible to do and you just answered the question :)

I understand that having this at multiple places is not convenient and I will deal with that accordingly.

Thanks!
